R/kernel.r

```diff
@@ -381,6 +381,20 @@ main <- function(connection_file = '') {
         connection_file <- commandArgs(TRUE)[[1]]
     }
     log_debug('Starting the R kernel...')
+    # Environment which will take functions which shadow main R stuff like q()
+    # This has to be created here, because a) it didn't work when I tried to add
+    # it as a normal package item and not here in the codepath of the startup
+    # function and b) so the on.exit can detach the env when main is finished and
+    # c) it can't be created elsewhere because attach copies the env, so it would
+    # need to be reassigned afterwards...
```

Comment:
I don't fully follow all this, but putting it in the `main` function feels like the wrong place. Can it be set up in the constructor of the Executor class?